version: "3.8"
services:
  pihole:
    image: pihole/pihole:2022.04.2
    hostname: pihole.devantler.com
    environment:
      VIRTUAL_HOST: pihole.${DOMAIN:?}
      TZ: "Europe/Copenhagen"
      PIHOLE_DNS_: 1.1.1.1
      DNSMASQ_LISTENING: all
      WEBPASSWORD: ""
    volumes:
      - pihole_etc_pihole:/etc/pihole
      - pihole_etc_dnsmasq.d:/etc/dnsmasq.d
    ports:
      - 53:53/udp
      - 53:53/tcp
    networks:
      - public
    deploy:
      placement:
        constraints:
          - node.hostname == manager1
      labels:
        traefik.enable: "true"
        traefik.http.routers.pihole.service: "pihole"
        traefik.http.routers.pihole.rule: "Host(`pihole.${DOMAIN:?}`)"
        traefik.http.routers.pihole.entrypoints: "websecure"
        traefik.http.routers.pihole.tls.certresolver: "letsencrypt"
        traefik.http.routers.pihole.middlewares: pihole-admin-prefix@docker, pihole-auth@docker
        traefik.http.middlewares.pihole-admin-prefix.addprefix.prefix: /admin
        traefik.http.middlewares.pihole-auth.forwardauth.address: http://cloudflare-auth:8080/auth/${CLOUDFLARE_AUTH_AUD:?}
        traefik.http.services.pihole.loadbalancer.server.port: 80

networks:
  public:
    external: true

volumes:
  pihole_etc_pihole:
  pihole_etc_dnsmasq.d:
